server/src/main/java/online/dipa/hub/persistence/repositories/PlanTemplateRepository.java

```diff
@@ -1,8 +1,38 @@
 package online.dipa.hub.persistence.repositories;
 
+import static org.hibernate.jpa.QueryHints.HINT_CACHEABLE;
+
+import java.util.Collection;
+import java.util.Optional;
+
+import javax.persistence.QueryHint;
+
 import org.springframework.data.jpa.repository.JpaRepository;
+import org.springframework.data.jpa.repository.Query;
+import org.springframework.data.jpa.repository.QueryHints;
 
 import online.dipa.hub.persistence.entities.PlanTemplateEntity;
+import online.dipa.hub.persistence.entities.ProjectApproachEntity;
 
 public interface PlanTemplateRepository extends JpaRepository<PlanTemplateEntity, Long> {
+
+    @QueryHints(value = { @QueryHint(name = HINT_CACHEABLE, value = "true") })
+    @Query("from PlanTemplateEntity as template join template.projectApproaches as approach " +
+            "where approach = :projectApproach")
+    Optional<Collection<PlanTemplateEntity>> findByProjectApproach(ProjectApproachEntity projectApproach);
+
+    @QueryHints(value = { @QueryHint(name = HINT_CACHEABLE, value = "true") })
+    @Query("from PlanTemplateEntity as template join template.projectApproaches as approach " +
+            "where template.standard = true and approach = :projectApproach")
+    Optional<PlanTemplateEntity> findByStandardAndProjectApproach(ProjectApproachEntity projectApproach);
+
+    @QueryHints(value = { @QueryHint(name = HINT_CACHEABLE, value = "true") })
+    @Query("from PlanTemplateEntity as template join template.projectApproaches as approach " +
+            "where template.defaultTemplate = true and approach = :projectApproach")
+    Optional<PlanTemplateEntity> findByDefaultAndProjectApproach(ProjectApproachEntity projectApproach);
+
+    @QueryHints(value = { @QueryHint(name = HINT_CACHEABLE, value = "true") })
+    @Query("from PlanTemplateEntity as template where template.id = :templateId")
+    Optional<PlanTemplateEntity> findByTemplateId(Long templateId);
```

Comment:
In der Vererbungshierachie von `JpaRepository` gibt es bereits die folgenden Methoden (Unterschied: die eine liefert Optional, die andere wirft EntityNotFound):

```java
	/**
	 * Returns a reference to the entity with the given identifier. Depending on how the JPA persistence provider is
	 * implemented this is very likely to always return an instance and throw an
	 * {@link javax.persistence.EntityNotFoundException} on first access. Some of them will reject invalid identifiers
	 * immediately.
	 *
	 * @param id must not be {@literal null}.
	 * @return a reference to the entity with the given identifier.
	 * @see EntityManager#getReference(Class, Object) for details on when an exception is thrown.
	 * @since 2.5
	 */
	T getById(ID id);


	/**
	 * Retrieves an entity by its id.
	 *
	 * @param id must not be {@literal null}.
	 * @return the entity with the given id or {@literal Optional#empty()} if none found.
	 * @throws IllegalArgumentException if {@literal id} is {@literal null}.
	 */
	Optional<T> findById(ID id);
```

Reply:
es lag an einem falsch gesetzt OneToOne Beziehung, deswegen konnte ich hier findById nicht verwenden. Den Fehler wurde behoben :)